### ISSUE TITLE
Get-AzStorageAccount hangs unpredictably

### ISSUE THREAD

**Issue by amerghazal** [Azure/azure-powershell#10601]
## Description

When making the `Get-AzStorageAccount` on both 2.6.0 and 3.0, we've found that unpredictably (once every ~100-150 runs) it hangs. By hangs, I mean the Powershell terminal becomes unusable and the process cannot be Ctrl-C'd. 

## Steps to reproduce

```powershell
0..100 | % { Write-Host "Iteration $_"; Get-AzStorageAccount -ResourceGroupName 'file-integration-test-fixture' -Name 'fileintegrationtest' }
```

## Environment data

<!-- Please run $PSVersionTable and paste the output in the below code block -->

```

Name                           Value                                                                                                                                                                                                                                                                                                                                                                                               
----                           -----                                                                                                                                                                                                                                                                                                                                                                                               
PSVersion                      5.1.16299.1146                                                                                                                                                                                                                                                                                                                                                                                      
PSEdition                      Desktop                                                                                                                                                                                                                                                                                                                                                                                             
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                                                                                                                                                                                             
BuildVersion                   10.0.16299.1146                                                                                                                                                                                                                                                                                                                                                                                     
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                                                                                                                                                                                     
WSManStackVersion              3.0                                                                                                                                                                                                                                                                                                                                                                                                 
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                                                                                                                                                                                                 
SerializationVersion           1.1.0.1                                                                                                                                                                                                                                                                                                                                                                                             

```

## Module versions

<!-- Please run (Get-Module  -ListAvailable) and paste the output in the below code block -->

```powershell

    Directory: C:\Program Files\WindowsPowerShell\Modules


ModuleType Version    Name                                ExportedCommands                                                                                                                                                                                                                                                                                                                                                         
---------- -------    ----                                ----------------                                                                                                                                                                                                                                                                                                                                                         
Script     1.6.4      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave...}                                                                                                                                                                                                                                                              
Script     1.6.2      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave...}                                                                                                                                                                                                                                                              
Script     1.1.0      Az.Advisor                          {Get-AzAdvisorRecommendation, Enable-AzAdvisorRecommendation, Disable-AzAdvisorRecommendation, Get-AzAdvisorConfiguration...}                                                                                                                                                                                                                                            
Script     1.0.1      Az.Advisor                          {Get-AzAdvisorRecommendation, Enable-AzAdvisorRecommendation, Disable-AzAdvisorRecommendation, Get-AzAdvisorConfiguration...}                                                                                                                                                                                                                                            
Script     1.0.2      Az.Aks                              {Get-AzAks, New-AzAks, Remove-AzAks, Import-AzAksCredential...}                                                                                                                                                                                                                                                                                                          
Script     1.1.1      Az.AnalysisServices                 {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysisServicesServer...}                                                                                                                                                                                                                                    
Script     1.3.2      Az.ApiManagement                    {Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion, Add-AzApiManagementUserToGroup...}                                                                                                                                                                                                                                       
Script     1.3.0      Az.ApiManagement                    {Add-AzApiManagementApiToProduct, Add-AzApiManagementProductToGroup, Add-AzApiManagementRegion, Add-AzApiManagementUserToGroup...}                                                                                                                                                                                                                                       
Script     1.0.2      Az.ApplicationInsights              {Get-AzApplicationInsights, New-AzApplicationInsights, Remove-AzApplicationInsights, Set-AzApplicationInsightsPricingPlan...}                                                                                                                                                                                                                                            
Script     1.3.4      Az.Automation                       {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-AzAutomationDscNodeConfiguration...}                                                                                                                                                                                                                   
Script     1.3.2      Az.Automation                       {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybridWorkerGroup, Get-AzAutomationJobOutputRecord, Import-AzAutomationDscNodeConfiguration...}                                                                                                                                                                                                                   
Script     2.0.1      Az.Batch                            {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKey, New-AzBatchAccount...}                                                                                                                                                                                                                                                                                
Script     1.1.1      Az.Batch                            {Remove-AzBatchAccount, Get-AzBatchAccount, Get-AzBatchAccountKey, New-AzBatchAccount...}                                                                                                                                                                                                                                                                                
Script     1.0.1      Az.Billing                          {Get-AzBillingInvoice, Get-AzBillingPeriod, Get-AzEnrollmentAccount, Get-AzConsumptionBudget...}                                                                                                                                                                                                                                                                         
Script     1.4.0      Az.Cdn                              {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-AzCdnProfile, Remove-AzCdnProfile...}                                                                                                                                                                                                                                                                                     
Script     1.3.1      Az.Cdn                              {Get-AzCdnProfile, Get-AzCdnProfileSsoUrl, New-AzCdnProfile, Remove-AzCdnProfile...}                                                                                                                                                                                                                                                                                     
Script     1.2.1      Az.CognitiveServices                {Get-AzCognitiveServicesAccount, Get-AzCognitiveServicesAccountKey, Get-AzCognitiveServicesAccountSku, Get-AzCognitiveServicesAccountType...}                                                                                                                                                                                                                            
Script     3.0.0      Az.Compute                          {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySet...}                                                                                                                                                                                                                                                                    
Script     2.5.0      Az.Compute                          {Remove-AzAvailabilitySet, Get-AzAvailabilitySet, New-AzAvailabilitySet, Update-AzAvailabilitySet...}                                                                                                                                                                                                                                                                    
Script     1.0.1      Az.ContainerInstance                {New-AzContainerGroup, Get-AzContainerGroup, Remove-AzContainerGroup, Get-AzContainerInstanceLog}                                                                                                                                                                                                                                                                        
Script     1.1.0      Az.ContainerRegistry                {New-AzContainerRegistry, Get-AzContainerRegistry, Update-AzContainerRegistry, Remove-AzContainerRegistry...}                                                                                                                                                                                                                                                            
Script     1.4.1      Az.DataFactory                      {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Get-AzDataFactoryV2, Remove-AzDataFactoryV2...}                                                                                                                                                                                                                                                                            
Script     1.2.0      Az.DataFactory                      {Set-AzDataFactoryV2, Update-AzDataFactoryV2, Get-AzDataFactoryV2, Remove-AzDataFactoryV2...}                                                                                                                                                                                                                                                                            
Script     1.0.1      Az.DataLakeAnalytics                {Get-AzDataLakeAnalyticsDataSource, New-AzDataLakeAnalyticsCatalogCredential, Remove-AzDataLakeAnalyticsCatalogCredential, Set-AzDataLakeAnalyticsCatalogCredential...}                                                                                                                                                                                                  
Script     1.2.4      Az.DataLakeStore                    {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule, Set-AzDataLakeStoreTrustedIdProvider...}                                                                                                                                                                                                             
Script     1.2.2      Az.DataLakeStore                    {Get-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreTrustedIdProvider, Remove-AzDataLakeStoreFirewallRule, Set-AzDataLakeStoreTrustedIdProvider...}                                                                                                                                                                                                             
Script     1.0.1      Az.DeploymentManager                {Get-AzDeploymentManagerArtifactSource, New-AzDeploymentManagerArtifactSource, Set-AzDeploymentManagerArtifactSource, Remove-AzDeploymentManagerArtifactSource...}                                                                                                                                                                                                       
Script     1.0.0      Az.DevTestLabs                      {Get-AzDtlAllowedVMSizesPolicy, Get-AzDtlAutoShutdownPolicy, Get-AzDtlAutoStartPolicy, Get-AzDtlVMsPerLabPolicy...}                                                                                                                                                                                                                                                      
Script     1.1.1      Az.Dns                              {Get-AzDnsRecordSet, New-AzDnsRecordConfig, Remove-AzDnsRecordSet, Set-AzDnsRecordSet...}                                                                                                                                                                                                                                                                                
Script     1.2.2      Az.EventGrid                        {New-AzEventGridTopic, Get-AzEventGridTopic, Set-AzEventGridTopic, New-AzEventGridTopicKey...}                                                                                                                                                                                                                                                                           
Script     1.4.0      Az.EventHub                         {New-AzEventHubNamespace, Get-AzEventHubNamespace, Set-AzEventHubNamespace, Remove-AzEventHubNamespace...}                                                                                                                                                                                                                                                               
Script     1.1.2      Az.FrontDoor                        {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDoor, Remove-AzFrontDoor...}                                                                                                                                                                                                                                                                                               
Script     1.1.1      Az.FrontDoor                        {New-AzFrontDoor, Get-AzFrontDoor, Set-AzFrontDoor, Remove-AzFrontDoor...}                                                                                                                                                                                                                                                                                               
Script     3.0.0      Az.HDInsight                        {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefinition, Wait-AzHDInsightJob, New-AzHDInsightStreamingMapReduceJobDefinition...}                                                                                                                                                                                                                                          
Script     2.0.1      Az.HDInsight                        {Get-AzHDInsightJob, New-AzHDInsightSqoopJobDefinition, Wait-AzHDInsightJob, New-AzHDInsightStreamingMapReduceJobDefinition...}                                                                                                                                                                                                                                          
Script     1.0.0      Az.HealthcareApis                   {New-AzHealthcareApisService, Remove-AzHealthcareApisService, Set-AzHealthcareApisService, Get-AzHealthcareApisService}                                                                                                                                                                                                                                                  
Script     2.0.0      Az.IotHub                           {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJob...}                                                                                                                                                                                                                                                                   
Script     1.2.1      Az.IotHub                           {Add-AzIotHubKey, Get-AzIotHubEventHubConsumerGroup, Get-AzIotHubConnectionString, Get-AzIotHubJob...}                                                                                                                                                                                                                                                                   
Script     1.3.1      Az.KeyVault                         {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, Stop-AzKeyVaultCertificateOperation, Get-AzKeyVaultCertificateOperation...}                                                                                                                                                                                                                                    
Script     1.3.1      Az.LogicApp                         {Get-AzIntegrationAccountAgreement, Get-AzIntegrationAccountAssembly, Get-AzIntegrationAccountBatchConfiguration, Get-AzIntegrationAccountCallbackUrl...}                                                                                                                                                                                                                
Script     1.1.1      Az.MachineLearning                  {Move-AzMlCommitmentAssociation, Get-AzMlCommitmentAssociation, Get-AzMlCommitmentPlanUsageHistory, Remove-AzMlCommitmentPlan...}                                                                                                                                                                                                                                        
Script     1.0.1      Az.ManagedServices                  {Get-AzManagedServicesAssignment, New-AzManagedServicesAssignment, Remove-AzManagedServicesAssignment, Get-AzManagedServicesDefinition...}                                                                                                                                                                                                                               
Script     1.0.1      Az.MarketplaceOrdering              {Get-AzMarketplaceTerms, Set-AzMarketplaceTerms}                                                                                                                                                                                                                                                                                                                         
Script     1.1.0      Az.Media                            {Sync-AzMediaServiceStorageKey, Set-AzMediaServiceKey, Get-AzMediaServiceKey, Get-AzMediaServiceNameAvailability...}                                                                                                                                                                                                                                                     
Script     1.4.0      Az.Monitor                          {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfile...}                                                                                                                                                                                                                                                                                         
Script     1.2.2      Az.Monitor                          {Get-AzMetricDefinition, Get-AzMetric, Remove-AzLogProfile, Get-AzLogProfile...}                                                                                                                                                                                                                                                                                         
Script     2.0.0      Az.Network                          {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplicationGatewayAuthenticationCertificate, Remove-AzApplicationGatewayAuthenticationCertificate...}                                                                                                                                                       
Script     1.13.0     Az.Network                          {Add-AzApplicationGatewayAuthenticationCertificate, Get-AzApplicationGatewayAuthenticationCertificate, New-AzApplicationGatewayAuthenticationCertificate, Remove-AzApplicationGatewayAuthenticationCertificate...}                                                                                                                                                       
Script     1.1.0      Az.NotificationHubs                 {Get-AzNotificationHub, Get-AzNotificationHubAuthorizationRule, Get-AzNotificationHubListKey, Get-AzNotificationHubPNSCredential...}                                                                                                                                                                                                                                     
Script     1.3.3      Az.OperationalInsights              {New-AzOperationalInsightsAzureActivityLogDataSource, New-AzOperationalInsightsCustomLogDataSource, Disable-AzOperationalInsightsLinuxCustomLogCollection, Disable-AzOperationalInsightsIISLogCollection...}                                                                                                                                                             
Script     1.1.3      Az.PolicyInsights                   {Get-AzPolicyEvent, Get-AzPolicyState, Get-AzPolicyStateSummary, Get-AzPolicyRemediation...}                                                                                                                                                                                                                                                                             
Script     1.1.0      Az.PowerBIEmbedded                  {Remove-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollection, Get-AzPowerBIWorkspaceCollectionAccessKey, Get-AzPowerBIWorkspace...}                                                                                                                                                                                                                            
Script     1.0.0      Az.PrivateDns                       {Get-AzPrivateDnsZone, Remove-AzPrivateDnsZone, Set-AzPrivateDnsZone, New-AzPrivateDnsZone...}                                                                                                                                                                                                                                                                           
Script     2.0.1      Az.RecoveryServices                 {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-AzRecoveryServicesVault...}                                                                                                                                                                                                                             
Script     1.4.4      Az.RecoveryServices                 {Get-AzRecoveryServicesBackupProperty, Get-AzRecoveryServicesVault, Get-AzRecoveryServicesVaultSettingsFile, New-AzRecoveryServicesVault...}                                                                                                                                                                                                                             
Script     1.1.1      Az.RedisCache                       {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePatchSchedule...}                                                                                                                                                                                                                                       
Script     1.1.0      Az.RedisCache                       {Remove-AzRedisCachePatchSchedule, New-AzRedisCacheScheduleEntry, Get-AzRedisCachePatchSchedule, New-AzRedisCachePatchSchedule...}                                                                                                                                                                                                                                       
Script     1.0.2      Az.Relay                            {New-AzRelayNamespace, Get-AzRelayNamespace, Set-AzRelayNamespace, Remove-AzRelayNamespace...}                                                                                                                                                                                                                                                                           
Script     1.7.1      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment...}                                                                                                                                                                                                                                                                        
Script     1.6.2      Az.Resources                        {Get-AzProviderOperation, Remove-AzRoleAssignment, Get-AzRoleAssignment, New-AzRoleAssignment...}                                                                                                                                                                                                                                                                        
Script     1.4.0      Az.ServiceBus                       {New-AzServiceBusNamespace, Get-AzServiceBusNamespace, Set-AzServiceBusNamespace, Remove-AzServiceBusNamespace...}                                                                                                                                                                                                                                                       
Script     2.0.0      Az.ServiceFabric                    {Add-AzServiceFabricClientCertificate, Add-AzServiceFabricClusterCertificate, Add-AzServiceFabricNode, Add-AzServiceFabricNodeType...}                                                                                                                                                                                                                                   
Script     1.1.2      Az.ServiceFabric                    {Add-AzServiceFabricApplicationCertificate, Add-AzServiceFabricClientCertificate, Add-AzServiceFabricClusterCertificate, Add-AzServiceFabricNode...}                                                                                                                                                                                                                     
Script     1.1.0      Az.SignalR                          {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKey...}                                                                                                                                                                                                                                                                                                    
Script     1.0.3      Az.SignalR                          {New-AzSignalR, Get-AzSignalR, Get-AzSignalRKey, New-AzSignalRKey...}                                                                                                                                                                                                                                                                                                    
Script     2.0.0      Az.Sql                              {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseUpgradeHint...}                                                                                                                                                                                            
Script     1.14.1     Az.Sql                              {Get-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseTransparentDataEncryptionActivity, Set-AzSqlDatabaseTransparentDataEncryption, Get-AzSqlDatabaseUpgradeHint...}                                                                                                                                                                                            
Script     1.9.0      Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKey...}                                                                                                                                                                                                                                                                        
Script     1.6.0      Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStorageAccount, New-AzStorageAccountKey...}                                                                                                                                                                                                                                                                        
Script     1.2.1      Az.StorageSync                      {Invoke-AzStorageSyncCompatibilityCheck, New-AzStorageSyncService, Get-AzStorageSyncService, Remove-AzStorageSyncService...}                                                                                                                                                                                                                                             
Script     1.2.0      Az.StorageSync                      {Invoke-AzStorageSyncCompatibilityCheck, New-AzStorageSyncService, Get-AzStorageSyncService, Remove-AzStorageSyncService...}                                                                                                                                                                                                                                             
Script     1.0.0      Az.StreamAnalytics                  {Get-AzStreamAnalyticsFunction, Get-AzStreamAnalyticsDefaultFunctionDefinition, New-AzStreamAnalyticsFunction, Remove-AzStreamAnalyticsFunction...}                                                                                                                                                                                                                      
Script     1.0.2      Az.TrafficManager                   {Add-AzTrafficManagerCustomHeaderToEndpoint, Remove-AzTrafficManagerCustomHeaderFromEndpoint, Add-AzTrafficManagerCustomHeaderToProfile, Remove-AzTrafficManagerCustomHeaderFromProfile...}                                                                                                                                                                              
Script     1.5.0      Az.Websites                         {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlan...}                                                                                                                                                                                                                                                                           
Script     1.4.1      Az.Websites                         {Get-AzAppServicePlan, Set-AzAppServicePlan, New-AzAppServicePlan, Remove-AzAppServicePlan...}                                                                                                                                                                                                                                                                           
Script     1.0.43     Communary.PASM                      {Select-ApproximateString, Select-SoundexString, Select-FuzzyString, Compare-PasmAlgorithms...}                                                                                                                                                                                                                                                                          
Script     1.0.0      Git.Management                      {Get-RemoteBranch, Where-FeatureBranch, Remove-RemoteBranch, Out-BranchReport}                                                                                                                                                                                                                                                                                           
Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}                                                                                                                                                                                                                                                                                                                    
Manifest   1.7.3      Ops.Infrastructure.Deployment       {Confirm-ChefNodesCheckingIn, Connect-IntoAzure, Connect-IntoAzureAD, Connect-IntoSubscription...}                                                                                                                                                                                                                                                                       
Manifest   1.2.2      Ops.Infrastructure.Deployment       {Confirm-ChefNodesCheckingIn, Connect-IntoAzure, Connect-IntoAzureAD, Connect-IntoSubscription...}                                                                                                                                                                                                                                                                       
Script     1.1.7.2    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}                                                                                                                                                                                                                                                                                                   
Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}                                                                                                                                                                                                                                                                                                   
Script     4.3.1      Pester                              {Describe, Context, It, Should...}                                                                                                                                                                                                                                                                                                                                       
Script     3.4.0      Pester                              {Describe, Context, It, Should...}                                                                                                                                                                                                                                                                                                                                       
Script     0.7.1      posh-git                            {Invoke-NullCoalescing, Add-PoshGitToProfile, Write-GitStatus, Write-Prompt...}                                                                                                                                                                                                                                                                                          
Script     2.0.1      PowerShellGet                       {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapability...}                                                                                                                                                                                                                                                                                                    
Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}                                                                                                                                                                                                                                                                                                             
Script     0.5.0      PSBuildTools                        {Import-RequiredAzureRmModules, Save-BuildArtifacts, Set-DigitalSignature, Set-LegacyDateBasedBuildVersion}                                                                                                                                                                                                                                                              
Script     0.2.26     PSCodeHealth                        {Invoke-PSCodeHealth, Get-PSCodeHealthComplianceRule, Test-PSCodeHealthCompliance, ipch}                                                                                                                                                                                                                                                                                 
Script     1.2        PSReadline                          {Get-PSReadlineKeyHandler, Set-PSReadlineKeyHandler, Remove-PSReadlineKeyHandler, Get-PSReadlineOption...}                                                                                                                                                                                                                                                               
Script     1.17.1     PSScriptAnalyzer                    {Get-ScriptAnalyzerRule, Invoke-ScriptAnalyzer, Invoke-Formatter}                                                                                                                                                                                                                                                                                                        
Script     1.17.2     R1Cmdb                              {Get-R1CmdbAbbreviationFromRegion, Get-R1CmdbChefServer, Get-R1CmdbIsProduction, Get-R1CmdbPod...}                                                                                                                                                                                                                                                                       
Script     1.17.1     R1Cmdb                              {Get-R1CmdbAbbreviationFromRegion, Get-R1CmdbChefServer, Get-R1CmdbIsProduction, Get-R1CmdbPod...}                                                                                                                                                                                                                                                                       
Script     1.3.1      R1Cmdb                              {Get-R1CmdbAbbreviationFromRegion, Get-R1CmdbChefServer, Get-R1CmdbIsProduction, Get-R1CmdbPod...}                                                                                                                                                                                                                                                                       
Script     3.0.0      R1Ops                                                                                                                                                                                                                                                                                                                                                                                                        
Script     0.3.2      R1Ops.Chef                          {Get-R1OpsChefNodeLastCheckIn, Get-R1OpsChefNodeRunlist, Set-R1OpsChefNodeRunlist, Test-R1OpsChefRecipeExists...}                                                                                                                                                                                                                                                        
Script     2.0.0      R1Ops.ChefEnvironment               {Get-R1OpsChefEnvironment, Get-R1OpsChefEnvironmentAudit, Update-R1OpsChefEnvironment, Remove-R1OpsChefEnvironment...}                                                                                                                                                                                                                                                   
Script     1.3.2      R1Ops.ChefPushJobs                  {Invoke-R1OpsPushJob, Get-R1OpsPushJobStatus}                                                                                                                                                                                                                                                                                                                            
Script     1.0.0      R1Ops.ChefServer                    {Get-R1OpsChefServerContext, Get-R1OpsChefServerInfo, Remove-R1OpsChefAutomateClientRun, Set-R1OpsChefServerContext...}                                                                                                                                                                                                                                                  
Script     2.0.0      R1Ops.Core                          {Connect-R1OpsAzure, Invoke-R1OpsRetry, Merge-R1OpsHashtable, ConvertTo-R1OpsHashtable...}                                                                                                                                                                                                                                                                               
Script     1.0.0      R1Ops.PodService                    {Get-R1OpsPod, Set-R1OpsPodContext}                                                                                                                                                                                                                                                                                                                                      
Script     1.1.0      R1Ops.Relativity                    {Get-R1OpsInstanceSetting, Invoke-R1OpsKepler, New-R1OpsInstanceSetting, Set-R1OpsInstanceSetting...}                                                                                                                                                                                                                                                                    
Script     2.0.0      R1Ops.SharedImage                   {Get-R1OpsImage, New-R1OpsSharedImageGalleryVersion, Get-R1OpsSharedImageGalleryVersion, Update-R1OpsSharedImageGalleryVersionRegions}                                                                                                                                                                                                                                   
Script     2.0.0      R1Ops.SharedSQL                     {Get-R1OpsSharedSQLHive, Get-R1OpsSharedSQLYard, Set-R1OpsSharedSQLContext, Get-R1OpsTenantSQLPaaSInstance}                                                                                                                                                                                                                                                              
Script     2.0.0      R1Ops.Storage                       {Get-R1OpsStorageBlobContent, Invoke-R1OpsDownloadRap}                                                                                                                                                                                                                                                                                                                   
Script     3.0.1      R1Ops.TenantService                 {Connect-R1OpsTenantSubscription, Get-R1OpsRelativityConnectionInfo, Get-R1OpsTenantEnvironment, Get-R1OpsTenantEnvironmentNames...}                                                                                                                                                                                                                                     
Script     2.0.0      R1Ops.Utilities                     {Get-CurrentTenants, Get-EnvironmentDetails, Invoke-R1OpsKnifeCommand}                                                                                                                                                                                                                                                                                                   
Manifest   1.0.1      RAPTools                            {Get-RelativityApplicationInstallationStatus, Get-RelativityLibraryApplication, Install-RelativityLibraryApplication, Start-RelativityWorkspaceApplicationUpgrade...}                                                                                                                                                                                                    
Binary     4.0.1      TvmPowershellModule                 {Get-TestVm, Select-TestVm, Find-TestVm, Install-TestVm...}                                                                                                                                                                                                                                                                                                              
Binary     3.1.0      TvmPowershellModule                 {Get-TestVm, Select-TestVm, Find-TestVm, Install-TestVm...}                                                                                                                                                                                                                                                                                                              
Binary     3.0.1      TvmPowershellModule                 {Get-TestVm, Select-TestVm, Find-TestVm, Install-TestVm...}                                                                                                                                                                                                                                                                                                              
Manifest   2.2.0      TvmPowershellModule                 {Get-TestVm, Find-TestVm, Install-TestVm, Remove-TestVm...}                                                                                                                                                                                                                                                                                                              


    Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules


ModuleType Version    Name                                ExportedCommands                                                                                                                                                                                                                                                                                                                                                         
---------- -------    ----                                ----------------                                                                                                                                                                                                                                                                                                                                                         
Manifest   1.0.0.0    AppBackgroundTask                   {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgroundTaskDiagnosticLog, Set-AppBackgroundTaskResourcePolicy, Unregister-AppBackgroundTask...}                                                                                                                                                                                                                    
Manifest   2.0.0.0    AppLocker                           {Get-AppLockerFileInformation, Get-AppLockerPolicy, New-AppLockerPolicy, Set-AppLockerPolicy...}                                                                                                                                                                                                                                                                         
Manifest   1.0.0.0    AppvClient                          {Add-AppvClientConnectionGroup, Add-AppvClientPackage, Add-AppvPublishingServer, Disable-Appv...}                                                                                                                                                                                                                                                                        
Manifest   2.0.0.0    Appx                                {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest, Remove-AppxPackage...}                                                                                                                                                                                                                                                                                       
Script     1.0.0.0    AssignedAccess                      {Clear-AssignedAccess, Get-AssignedAccess, Set-AssignedAccess}                                                                                                                                                                                                                                                                                                           
Manifest   1.0.0.0    BitLocker                           {Unlock-BitLocker, Suspend-BitLocker, Resume-BitLocker, Remove-BitLockerKeyProtector...}                                                                                                                                                                                                                                                                                 
Manifest   2.0.0.0    BitsTransfer                        {Add-BitsFile, Complete-BitsTransfer, Get-BitsTransfer, Remove-BitsTransfer...}                                                                                                                                                                                                                                                                                          
Manifest   1.0.0.0    BranchCache                         {Add-BCDataCacheExtension, Clear-BCCache, Disable-BC, Disable-BCDowngrading...}                                                                                                                                                                                                                                                                                          
Manifest   1.0.0.0    CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, Get-CimSession...}                                                                                                                                                                                                                                                                                            
Manifest   1.0        ConfigCI                            {Get-SystemDriver, New-CIPolicyRule, New-CIPolicy, Get-CIPolicy...}                                                                                                                                                                                                                                                                                                      
Manifest   1.0        Defender                            {Get-MpPreference, Set-MpPreference, Add-MpPreference, Remove-MpPreference...}                                                                                                                                                                                                                                                                                           
Manifest   1.0.0.0    DeliveryOptimization                {Get-DeliveryOptimizationStatus, Get-DeliveryOptimizationPerfSnap}                                                                                                                                                                                                                                                                                                       
Manifest   1.0.0.0    DirectAccessClientComponents        {Disable-DAManualEntryPointSelection, Enable-DAManualEntryPointSelection, Get-DAClientExperienceConfiguration, Get-DAEntryPointTableItem...}                                                                                                                                                                                                                             
Script     3.0        Dism                                {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-WindowsCapability, Add-WindowsImage...}                                                                                                                                                                                                                                                                              
Manifest   1.0.0.0    DnsClient                           {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get-DnsClientCache...}                                                                                                                                                                                                                                                                                            
Manifest   1.0.0.0    EventTracingManagement              {Start-EtwTraceSession, New-EtwTraceSession, Get-EtwTraceSession, Update-EtwTraceSession...}                                                                                                                                                                                                                                                                             
Manifest   1.0.0.0    HgsClient                           {Get-HgsAttestationBaselinePolicy, Get-HgsClientConfiguration, Test-HgsClientConfiguration, Set-HgsClientConfiguration...}                                                                                                                                                                                                                                               
Manifest   1.0.0.0    HgsDiagnostics                      {New-HgsTraceTarget, Get-HgsTrace, Get-HgsTraceFileData, Test-HgsTraceTarget}                                                                                                                                                                                                                                                                                            
Manifest   1.0.0.0    HostNetworkingService               {Get-HnsEndpoint, Remove-HNSNetwork, Get-HNSNetwork, Remove-HNSEndpoint}                                                                                                                                                                                                                                                                                                 
Binary     2.0.0.0    Hyper-V                             {Add-VMAssignableDevice, Add-VMDvdDrive, Add-VMFibreChannelHba, Add-VMGpuPartitionAdapter...}                                                                                                                                                                                                                                                                            
Binary     1.1        Hyper-V                             {Add-VMDvdDrive, Add-VMFibreChannelHba, Add-VMHardDiskDrive, Add-VMMigrationNetwork...}                                                                                                                                                                                                                                                                                  
Script     1.1.0.0    IISAdministration                   {Get-IISAppPool, Start-IISCommitDelay, Stop-IISCommitDelay, Get-IISSite...}                                                                                                                                                                                                                                                                                              
Manifest   2.0.0.0    International                       {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMethodOverride, Get-WinHomeLocation, Set-WinHomeLocation...}                                                                                                                                                                                                                                                      
Manifest   1.0.0.0    iSCSI                               {Get-IscsiTargetPortal, New-IscsiTargetPortal, Remove-IscsiTargetPortal, Update-IscsiTargetPortal...}                                                                                                                                                                                                                                                                    
Script     1.0.0.0    ISE                                 {New-IseSnippet, Import-IseSnippet, Get-IseSnippet}                                                                                                                                                                                                                                                                                                                      
Manifest   1.0.0.0    Kds                                 {Add-KdsRootKey, Get-KdsRootKey, Test-KdsRootKey, Set-KdsConfiguration...}                                                                                                                                                                                                                                                                                               
Manifest   1.0.1.0    Microsoft.PowerShell.Archive        {Compress-Archive, Expand-Archive}                                                                                                                                                                                                                                                                                                                                       
Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    {Get-WinEvent, Get-Counter, Import-Counter, Export-Counter...}                                                                                                                                                                                                                                                                                                           
Manifest   3.0.0.0    Microsoft.PowerShell.Host           {Start-Transcript, Stop-Transcript}                                                                                                                                                                                                                                                                                                                                      
Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser, Get-LocalGroup...}                                                                                                                                                                                                                                                                                           
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path...}                                                                                                                                                                                                                                                                                                           
Script     1.0        Microsoft.PowerShell.ODataUtils     Export-ODataEndpointProxy                                                                                                                                                                                                                                                                                                                                                
Manifest   3.0.0.0    Microsoft.PowerShell.Security       {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential...}                                                                                                                                                                                                                                                                                                                
Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Format-List, Format-Custom, Format-Table, Format-Wide...}                                                                                                                                                                                                                                                                                                               
Manifest   3.0.0.0    Microsoft.WSMan.Management          {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSSP, Set-WSManQuickConfig...}                                                                                                                                                                                                                                                                                   
Manifest   1.0        MMAgent                             {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgent...}                                                                                                                                                                                                                                                                                                           
Manifest   1.0.0.0    MsDtc                               {New-DtcDiagnosticTransaction, Complete-DtcDiagnosticTransaction, Join-DtcDiagnosticResourceManager, Receive-DtcDiagnosticTransaction...}                                                                                                                                                                                                                                
Manifest   2.0.0.0    NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-NetAdapterChecksumOffload, Disable-NetAdapterEncapsulatedPacketTaskOffload...}                                                                                                                                                                                                                                   
Manifest   1.0.0.0    NetConnection                       {Get-NetConnectionProfile, Set-NetConnectionProfile}                                                                                                                                                                                                                                                                                                                     
Manifest   1.0.0.0    NetEventPacketCapture               {New-NetEventSession, Remove-NetEventSession, Get-NetEventSession, Set-NetEventSession...}                                                                                                                                                                                                                                                                               
Manifest   2.0.0.0    NetLbfo                             {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTeam, Get-NetLbfoTeamMember...}                                                                                                                                                                                                                                                                                   
Manifest   1.0.0.0    NetNat                              {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMapping, Get-NetNatSession...}                                                                                                                                                                                                                                                                                   
Manifest   2.0.0.0    NetQos                              {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, New-NetQosPolicy}                                                                                                                                                                                                                                                                                              
Manifest   2.0.0.0    NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsecMainModeCryptoProposal, New-NetIPsecQuickModeCryptoProposal...}                                                                                                                                                                                                                                               
Manifest   1.0.0.0    NetSwitchTeam                       {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTeam, Rename-NetSwitchTeam...}                                                                                                                                                                                                                                                                                    
Manifest   1.0.0.0    NetTCPIP                            {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol, Get-NetIPv6Protocol...}                                                                                                                                                                                                                                                                                      
Manifest   1.0.0.0    NetWNV                              {Get-NetVirtualizationProviderAddress, Get-NetVirtualizationGlobal, Get-NetVirtualizationLookupRecord, Get-NetVirtualizationCustomerRoute...}                                                                                                                                                                                                                            
Manifest   1.0.0.0    NetworkConnectivityStatus           {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Reset-NCSIPolicyConfiguration, Set-NCSIPolicyConfiguration}                                                                                                                                                                                                                                                        
Manifest   1.0.0.0    NetworkSwitchManager                {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEthernetPort, Get-NetworkSwitchEthernetPort, Remove-NetworkSwitchEthernetPortIPAddress...}                                                                                                                                                                                                                       
Manifest   1.0.0.0    NetworkTransition                   {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfiguration, Disable-NetIPHttpsProfile, Disable-NetNatTransitionConfiguration...}                                                                                                                                                                                                                                  
Manifest   1.0.0.0    PcsvDevice                          {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restart-PcsvDevice...}                                                                                                                                                                                                                                                                                               
Manifest   1.0.0.0    PKI                                 {Add-CertificateEnrollmentPolicyServer, Export-Certificate, Export-PfxCertificate, Get-CertificateAutoEnrollmentPolicy...}                                                                                                                                                                                                                                               
Manifest   1.0.0.0    PnpDevice                           {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, Disable-PnpDevice}                                                                                                                                                                                                                                                                                              
Manifest   1.1        PrintManagement                     {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-PrintConfiguration...}                                                                                                                                                                                                                                                                                             
Binary     1.0.11     ProcessMitigations                  {Get-ProcessMitigation, Set-ProcessMitigation, ConvertTo-ProcessMitigationPolicy}                                                                                                                                                                                                                                                                                        
Script     3.0        Provisioning                        {Install-ProvisioningPackage, Export-ProvisioningPackage, Install-TrustedProvisioningCertificate, Export-Trace...}                                                                                                                                                                                                                                                       
Manifest   1.1        PSDesiredStateConfiguration         {Set-DscLocalConfigurationManager, Start-DscConfiguration, Test-DscConfiguration, Publish-DscConfiguration...}                                                                                                                                                                                                                                                           
Script     1.0.0.0    PSDiagnostics                       {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSManTrace, Enable-PSTrace...}                                                                                                                                                                                                                                                                                   
Binary     1.1.0.0    PSScheduledJob                      {New-JobTrigger, Add-JobTrigger, Remove-JobTrigger, Get-JobTrigger...}                                                                                                                                                                                                                                                                                                   
Manifest   2.0.0.0    PSWorkflow                          {New-PSWorkflowExecutionOption, New-PSWorkflowSession, nwsn}                                                                                                                                                                                                                                                                                                             
Manifest   1.0.0.0    PSWorkflowUtility                   Invoke-AsWorkflow                                                                                                                                                                                                                                                                                                                                                        
Manifest   1.0.0.0    ScheduledTasks                      {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledTask, Unregister-ScheduledTask...}                                                                                                                                                                                                                                                                              
Manifest   2.0.0.0    SecureBoot                          {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBootUEFI, Format-SecureBootUEFI...}                                                                                                                                                                                                                                                                               
Manifest   2.0.0.0    SmbShare                            {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbShareAccess...}                                                                                                                                                                                                                                                                                                   
Manifest   2.0.0.0    SmbWitness                          {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbw...}                                                                                                                                                                                                                                                                                                           
Manifest   1.0.0.0    StartLayout                         {Export-StartLayout, Import-StartLayout, Export-StartLayoutEdgeAssets, Get-StartApps}                                                                                                                                                                                                                                                                                    
Manifest   2.0.0.0    Storage                             {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add-PhysicalDisk, Add-StorageFaultDomain...}                                                                                                                                                                                                                                                                      
Manifest   2.0.0.0    TLS                                 {New-TlsSessionTicketKey, Enable-TlsSessionTicketKey, Disable-TlsSessionTicketKey, Export-TlsSessionTicketKey...}                                                                                                                                                                                                                                                        
Manifest   1.0.0.0    TroubleshootingPack                 {Get-TroubleshootingPack, Invoke-TroubleshootingPack}                                                                                                                                                                                                                                                                                                                    
Manifest   2.0.0.0    TrustedPlatformModule               {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-Tpm...}                                                                                                                                                                                                                                                                                                                     
Binary     2.1.639.0  UEV                                 {Clear-UevConfiguration, Clear-UevAppxPackage, Restore-UevBackup, Set-UevTemplateProfile...}                                                                                                                                                                                                                                                                             
Manifest   2.0.0.0    VpnClient                           {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnection, Get-VpnConnection...}                                                                                                                                                                                                                                                                                       
Manifest   1.0.0.0    Wdac                                {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsn...}                                                                                                                                                                                                                                                                                                            
Manifest   1.0.0.0    WebAdministration                   {Start-WebCommitDelay, Stop-WebCommitDelay, Get-WebConfigurationLock, Remove-WebConfigurationLock...}                                                                                                                                                                                                                                                                    
Manifest   1.0.0.0    WindowsDeveloperLicense             {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperLicense, Show-WindowsDeveloperLicenseRegistration}                                                                                                                                                                                                                                                              
Script     1.0        WindowsErrorReporting               {Enable-WindowsErrorReporting, Disable-WindowsErrorReporting, Get-WindowsErrorReporting}                                                                                                                                                                                                                                                                                 
Manifest   1.0.0.0    WindowsSearch                       {Get-WindowsSearchSetting, Set-WindowsSearchSetting}                                                                                                                                                                                                                                                                                                                     
Manifest   1.0.0.0    WindowsUpdate                       Get-WindowsUpdateLog                                                                                                                                                                                                                                                                                                                                                     
Manifest   1.0.0.2    WindowsUpdateProvider               {Get-WUAVersion, Get-WULastInstallationDate, Get-WULastScanSuccessDate, Get-WUIsPendingReboot...}                                                                                                                                                                                                                                                                        


    Directory: C:\Program Files (x86)\Microsoft SQL Server\120\Tools\PowerShell\Modules


ModuleType Version    Name                                ExportedCommands                                                                                                                                                                                                                                                                                                                                                         
---------- -------    ----                                ----------------                                                                                                                                                                                                                                                                                                                                                         
Manifest   1.0        SQLASCMDLETS                        {Add-RoleMember, Backup-ASDatabase, Invoke-ASCmd, Invoke-ProcessCube...}                                                                                                                                                                                                                                                                                                 
Manifest   1.0        SQLPS                               {Backup-SqlDatabase, Add-SqlAvailabilityDatabase, Add-SqlAvailabilityGroupListenerStaticIp, Disable-SqlAlwaysOn...}                                                                                                                                                                                                                                                      


    Directory: C:\opscode\chefdk\modules


ModuleType Version    Name                                ExportedCommands                                                                                                                                                                                                                                                                                                                                                         
---------- -------    ----                                ----------------                                                                                                                                                                                                                                                                                                                                                         
Script     0.0        chef                                {knife, chef-client, chef-service-manager, chef-shell...}                                                                                                                                                                                                                                                                                                                



```

## Debug output

<!-- Set $DebugPreference='Continue' before running the repro and paste the resulting debug stream in the below code block -->

```
S:\Sourcecode> $DebugPreference = 'Continue'
S:\Sourcecode> $sa = Get-AzStorageAccount -ResourceGroupName 'file-integration-test-fixture' -Name 'fileintegrationtest'
DEBUG: 10:04:33 AM - GetAzureStorageAccountCommand begin processing with ParameterSet 'AccountNameParameterSet'.
DEBUG: 10:04:33 AM - using account id 'name.redacted@domain.com'...
```

## Error output

<!-- Please run Resolve-AzError and paste the output in the below code block -->

```
S:\Sourcecode> $DebugPreference = 'Continue'
S:\Sourcecode> Resolve-AzError
DEBUG: 10:07:05 AM - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.
DEBUG: 10:07:05 AM - using account id 'jeremy.wright@relativity.com'...
WARNING: Breaking changes in the cmdlet 'Resolve-AzError' :
WARNING:  - The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use `Resolve-AzError` instead.
WARNING: NOTE : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other information on breaking changes in Azure PowerShell.
   HistoryId: -1
Message        : Exception calling "CompleteInput" with "3" argument(s): "Sequence contains no elements"
StackTrace     :    at System.Management.Automation.ExceptionHandlingOps.ConvertToMethodInvocationException(Exception exception, Type typeToThrow, String methodName, Int32 numArgs, MemberInfo memberInfo)
                    at CallSite.Target(Closure , CallSite , Type , String , Int32 , Hashtable )
                    at System.Dynamic.UpdateDelegates.UpdateAndExecute4[T0,T1,T2,T3,TRet](CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3)
                    at System.Management.Automation.Interpreter.DynamicInstruction`5.Run(InterpretedFrame frame)
                    at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
Exception      : System.Management.Automation.MethodInvocationException
InvocationInfo : {}
Line           :         return [System.Management.Automation.CommandCompletion]::CompleteInput(
Position       : At line:38 char:16
                 + ...      return [System.Management.Automation.CommandCompletion]::Complet ...
                 +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : -1
Message        : Sequence contains no elements
StackTrace     :    at System.Linq.Enumerable.Last[TSource](IEnumerable`1 source)
                    at System.Management.Automation.CompletionAnalysis.CreateCompletionContext(ExecutionContext executionContext)
                    at System.Management.Automation.CompletionAnalysis.GetResults(PowerShell powerShell, Int32& replacementIndex, Int32& replacementLength)
                    at System.Management.Automation.CommandCompletion.CompleteInputImpl(Ast ast, Token[] tokens, IScriptPosition positionOfCursor, Hashtable options)
                    at CallSite.Target(Closure , CallSite , Type , String , Int32 , Hashtable )
Exception      : System.InvalidOperationException
InvocationInfo : {}
Line           :         return [System.Management.Automation.CommandCompletion]::CompleteInput(
Position       : At line:38 char:16
                 + ...      return [System.Management.Automation.CommandCompletion]::Complet ...
                 +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : -1
DEBUG: AzureQoSEvent: CommandName - Resolve-AzError; IsSuccess - True; Duration - 00:00:00.0466098; Exception - ;
DEBUG: Finish sending metric.
DEBUG: 10:07:05 AM - ResolveError end processing.
DEBUG: 10:07:05 AM - ResolveError end processing.
```

**Comment by blueww** [comment]
@amerghazal

I have tried to repro the issue with Az.Storage 1.9.0, but not repro.

I use Powershell 5.1 (5.1.18362.145), and run the Get-AzStorageAccount 100, 200, 1000 times , for several rounds, but not repro. Powershell always can finish the following script successfully.
`0..1000 | % { Write-Host "Iteration $_"; Get-AzStorageAccount -ResourceGroupName $rgname -StorageAccountName $accountName  }`
You can see the test log: [1000.txt](https://github.com/Azure/azure-powershell/files/3890787/1000.txt)

From the Resolve-AzError, it seems the error is reported from Powershell framework, but not see the trace from Azure Powershell cmdlet code.

Besides that, since as you said "Powershell terminal becomes unusable and the process cannot be Ctrl-C'd", how did you run AResolve-AzError after the the hang happen?

**Comment by chinchalupa** [comment]
@blueww 

We were able to further isolate the issue and explain why you may not be seeing this issue. The problem occurs only for storage accounts in the 'CentralUS' region. We tested our script above on a storage account created in 'EastUS2' and had no issue. However, if the storage account was located in 'CentralUS' (with same region, resource group, subscription, tenant, etc.), we would observe the described bug.
Please try reproducing again by deploying a storage account in the 'CentralUS' region first, then trying the query another 1000 iterations.
We were also able to reproduce this bug and the regional workaround in the Azure cloud shell further distancing us from a local environment issue.

**Comment by blueww** [comment]
@chinchalupa 
I have created an account in 'CentralUS' region, and rerun Get-AzStorageAccount 1000 times on it, still not repro the issue.
See the test log: [1000.txt](https://github.com/Azure/azure-powershell/files/3895924/1000.txt)

Besides that, if you really concern the Get-AzStorageAccount can't be be Ctrl-C'd in this scenario, you might can try to run the cmdlet asynchronously.

```powershell
$task = Start-Job -ScriptBlock { Get-AzStorageAccount -ResourceGroupName $rgname -StorageAccountName $accountName }

# wait for the task finish, or just don't wait but just check the $task.State is "Running" or not.
Wait-Job $task

# See the output of the cmdlet
$task.ChildJobs[0].Output
```

**Comment by blueww** [comment]
@chinchalupa
Do you have any update or further questions for this issue?

**Comment by blueww** [comment]
@chinchalupa
I will close the issue since not get any update from you for a long time.
Feel free to contact us if you need any further assistance.

**Comment by wjdavis5** [comment]
Hello - I would like to get this reopened, as we are seeing the same problem. I can reproduce locally by using a similar method as the OP. However, this mainly bites in the context of Azure Dev Ops Pipelines. Sporadically we'll get a day were we experience Agent Exhaustion and we'll find that numerous agents are hung on a task that calls:
```
Get-AzStorageAccount -ResourceGroupName "$ResourceGroupName"
```

Consider this output:

```
PS D:\> 0..100| %{$(Measure-Command {Get-AzStorageAccount -ResourceGroupName redacted-east-test}).Milliseconds}
Get-AzStorageAccount: A task was canceled.
83
811
723
32
```

I just happen to get lucky in this case and the first iteration hung. The amount of time things can hang seem to be indeterminant. It can rang from seconds to minutes, to in some extreme cases, several hours (ADO Pipelines). 




Here are my modules:

```
PS D:\git\Ownership> Get-Module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     1.7.4                 Az.Accounts                         {Add-AzEnvironment, Clear-AzContext, Clear-AzDefault, Connect-A…
Script     1.13.0                Az.Resources                        {Add-AzADGroupMember, Export-AzResourceGroup, Get-AzADAppCreden…
Script     1.13.0                Az.Storage                          {Add-AzRmStorageContainerLegalHold, Add-AzStorageAccountManagem…
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Security       {ConvertFrom-SecureString, ConvertTo-SecureString, Get-Acl, Get…
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Manifest   7.0.0.0               Microsoft.WSMan.Management          {Connect-WSMan, Disable-WSManCredSSP, Disconnect-WSMan, Enable-…
Script     2.0.0                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSRead
```

**Comment by blueww** [comment]
@wjdavis5 

I can't repro this issue by run the following command.
The time to run the cmdlet can change with network status, server behavior, ... On my macine, each cmdlet runs: 15ms to more than 900ms.
```powershell
0..100| %{$(Measure-Command {Get-AzStorageAccount -ResourceGroupName $rgname}).Milliseconds}
```
If you don't want the hang happen, you can use following command to limit the time of run the command. Following command will report error when it runs > 10s. You can use a longer time for stability.
```powershell
Invoke-AsWorkflow -CommandName Get-AzStorageAccount -Parameter @{"ResourceGroupName"=$rgname} -PSRunningTimeoutSec 10
```

**Comment by wjdavis5** [comment]
There is obviously something strange going on. As I said we see this problem from time to time in ADO and it causes our agents to hang until we terminate the job.

**Comment by blueww** [comment]
To avoid the hang of you agent, would you like to use the suggested way in my before comments to limit the time to run the cmdlet? You can retry if the output is null (which mean the cmdlet not finish in the timeout). So your agent won't be hang. Let me know if you see any issue to use it.

Besides that, I have tried to run > 1000 times, still not repro this issue.
If you can give a network trace for the issue happen, that might can help to investigate this issue.

**Comment by wjdavis5** [comment]
Invoking as workflow is really a solution. I'm going to work on getting more detailed tracing info

**Comment by coatesgavin** [comment]
I think this needs to be re-opened. I'm seeing the same issue. I have a script that loops through all storage accounts to get info. I've seen persistent hanging on my runs since morning of 23/4. A workaround doesn't address the root cause.

`Name                           Value                                                                                                                                                                                                                                           
----                           -----                                                                                                                                                                                                                                           
PSVersion                      5.1.17763.771                                                                                                                                                                                                                                   
PSEdition                      Desktop                                                                                                                                                                                                                                         
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                                         
BuildVersion                   10.0.17763.771                                                                                                                                                                                                                                  
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                                 
WSManStackVersion              3.0                                                                                                                                                                                                                                             
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                                             
SerializationVersion           1.1.0.1      `

az.storage version is 1.9.0 however a colleague reproduced the issue on a later version.

debug output shows:
`DEBUG: 10:52:37 - using account id........ '`

Then it hangs. As mentioned previously it cannot be cancelled with ctrl-c.

On a successful run the next debug out expected is:
`DEBUG: [Common.Authentication]: Authenticating using Account: '.....', environment:
'AzureCloud', tenant: '.....'`

So it's tripping up before it returns this.

The location of the storage accounts is North Europe.

**Comment by AlexKeySmith** [comment]
Hi same here from a MS hosted agent and a storage account in UK South. In particular Get-AzRoleAssignment on an existing storage account, it hangs indefinitely. However it runs fine ok on a newly deployed storage account, when executed locally. 

I'll share some diagnostic details if I can get any where with it.

**Comment by blueww** [comment]
@coatesgavin 
Would you please give the repro script?  (hide any credential)
And if the hang on  `Get-AzRoleAssignment`  or `Get-AzStorageaccount`?

**Comment by AlexKeySmith** [comment]
Thanks for getting back to us so quickly @blueww 

I stand corrected, it was New instead.

$blobDelegatorArgs = @{
                        ObjectId = $principalId
                        ResourceGroupName = $ResourceGroupName
                        ResourceName = $deployment.Outputs.storageAccountName.value
                        RoleDefinitionName = "Storage Blob Delegator"
                        ResourceType = "Microsoft.Storage/storageAccounts"
                    }

At this point the storage account has been already been created by a resource group deployment.

New-AzRoleAssignment @blobDelegatorArgs

I have a meeting in a moment, but I'll try debug mode on the next run!

**Comment by blueww** [comment]
@AlexKeySmith 
As the hang is in 'New-AzRoleAssignment', it's actually a new issue.
Would you please open a new issue, and the proper team will help you on this.

Please follow the issue teamplate and include you repro script, debug info ... in the new issue.

**Comment by AlexKeySmith** [comment]
Ah yes fair point, thanks, of course it'll be different endpoints underneath. I'll raise a new issue, have a nice weekend.

**Comment by AlexKeySmith** [comment]
For those debugging similar things, my actual problem was serialising the exception for logging!
After spending some time creating a repro locally (I only saw it on our build agent originally). The problem was:

Write-Host ($_.Exception | ConvertTo-Json -Depth 10) 

Where I suspect there's a circular reference inside the .net types within Microsoft.Rest.Azure.CloudException, changing to -Depth 2 was ok.

**Comment by coatesgavin** [comment]
Hi, Apologies, i've had to work on something else recently and haven't had time to look at this.
It's a large script but essentially, for each subscription we manage we do the following:
$resources = Get-AzResource
$SAResources = resources | Where-Object{$_.ResourceType -eq "Microsoft.Storage/storageAccounts"}
ForEach($azureSA in $azureSAResources)
{
	$azureSAResource = Get-AzStorageAccount -ResourceGroupName $azureSA.ResourceGroupName -Name $azureSA.Name
}

We then pull out the required information but it's Get-AzStorageAccount where we see the hang.

We had a look at process explorer during the issue and found it was making calls out then the thread exits.

![2020-05-15 10_22_22-Photos](https://user-images.githubusercontent.com/49303354/82036324-ad01e780-9698-11ea-9715-20b08aa92b50.png)

Interestingly this script runs against multiple subscriptions but only one seems to be having this issue.

I wrapped the command in invoke-asworkflow but it persistently times out, only getting info back on a small amount of storage accounts.

Also it's not just Storage accounts, we use the same method on a load of other resources, storage accounts are the only ones we see the issue with.

**Comment by blueww** [comment]
@coatesgavin 
Thanks for the information!

We plan to make a change in the next release to allow Get-AzStorageAccount run Async with -Asjob parameter. You might can try if it can resolve you problem after the next release is out.

For the thread exits, not see a reason why it exists, and can't repro it. 
Do you have the network trace when the issue happen?
Or if the -Asjob can't resolve your issue, we can look deeper into it.

For invoke-asworkflow timeout and only get a small amout of storage account, can you set the timeout to longer?

**Comment by blueww** [comment]
@coatesgavin

The [latest Azure Powershell](https://github.com/Azure/azure-powershell/releases/tag/v4.1.0-May2020) is released. It support -asjob in Get-AzStorageAccount. 
Would you please try to modify `$azureSAResource = Get-AzStorageAccount -ResourceGroupName $azureSA.ResourceGroupName -Name $azureSA.Name` to the following script, to make the get account won't always wait and can retry on timeout to make it more stable. 
You can adjust the retryTimes and waitTimeInSecond according to your environment.

Let me know if it works for you or not.

```powershell
$retryTimes = 3
$waitTimeInSecond = 10

do {
    $job = Get-AzStorageAccount -ResourceGroupName $azureSA.ResourceGroupName -Name $azureSA.Name -AsJob
    $job | wait-job -Timeout $waitTimeInSecond  
} while (($retryTimes-- -gt 1) -and ($job.State -ne "Completed"))

if($job.State -eq "Completed")
{
   $azureSAResource  = $job.Output
}
else 
{
    # handle the list timeout error
}
```
